sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -903,15 +910,55 @@ private[joins] object LongHashedRelation {
       if (!rowKey.isNullAt(0)) {
         val key = rowKey.getLong(0)
         map.append(key, unsafeRow)
+      } else if (isNullAware) {
+        return new EmptyHashedRelationWithAllNullKeys
       }
     }
     map.optimize()
     new LongHashedRelation(numFields, map)
   }
 }
 
+/**
+ * Common trait with dummy implementation for NAAJ special HashedRelation
+ * EmptyHashedRelation
+ * EmptyHashedRelationWithAllNullKeys
+ */
+trait NullAwareHashedRelation extends HashedRelation with Externalizable {
+  override def get(key: InternalRow): Iterator[InternalRow] = null
```

Comment:
Should the default implementations of these methods throw some `UnsupportedOperationException` instead of silently getting a null pointer exception in the caller ?

Reply:
![image](https://user-images.githubusercontent.com/17242071/88567872-a9acb400-d06a-11ea-8334-1bb069f61f16.png)

from the HashedRelation definition, it is valid to return null, if not match found.
for a empty relation, i think it's ok to just return null for any input key. ^_^

Reply:
I agree it's safer to `UnsupportedOperationException`. They shouldn't be called and it's a bug if they are called, we should fail if bug happens.

Reply:
done updated